keel-sql/src/test/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepositoryTests.kt

```diff
@@ -30,6 +36,37 @@ internal object SqlResourceRepositoryTests : ResourceRepositoryTests<SqlResource
     cleanupDb(jooq)
   }
 
+  fun moreTests() = rootContext<Fixture<SqlResourceRepository>> {
+    fixture {
+      Fixture(subject = factory(clock))
+    }
+
+    after { confirmVerified(callback) }
+    after { flush() }
+
+    context("race condition between storing a resource and retrieving its uid") {
+      val resource = resource()
+      test("retries and succeeds retrieving resource uid") {
+        var exception: Throwable? = null
+        Thread {
+          Thread.sleep(200) // timeout is 5 retries * 100 millis
+          subject.store(resource)
+        }.start()
+        Thread {
+          try {
+            subject.getResourceUid(resource.id)
+          } catch (e: IllegalStateException) {
+            exception = e
+          }
+        }.also {
+          it.start()
+          it.join()
+          expectThat(exception).isNull()
```

Comment:
@robfletcher I tried to write this with Strikt's `runCatching` but the assertion error was supressed, I think because it happened in the thread. I also tried running the calls in coroutines instead of threads, but I couldn't make that work (delaying/sleeping in one coroutine never seemed to have an effect). 

If you have thoughts on how to write this in a different way, please let me know.